### PR TITLE
Fix image cpu loading

### PR DESCRIPTION
**Pull request by yinweisu**
*Issue #, if available:*
https://github.com/awslabs/autogluon/issues/1426

*Description of changes:*
Previous changes in GluonCV only solved the estimator loading. Still need to handle the pickle logic. Added a cpu unpickler to handle the special logic. Refer to: https://github.com/pytorch/pytorch/issues/16797

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

**Comment by szha** [comment]
Job PR-1533-1 is done. 
Docs are uploaded to http://autogluon-staging.s3-website-us-west-2.amazonaws.com/PR-1533/1/index.html

File: vision/src/autogluon/vision/utils/pickle.py [hunk 1]
```diff
@@ -0,0 +1,12 @@
+import io
+import pickle
+
+
+# https://github.com/pytorch/pytorch/issues/16797
+class CPU_Unpickler(pickle.Unpickler):
```
Comment (Innixma) [inline vision/src/autogluon/vision/utils/pickle.py:6]: nit: Perhaps `CPUUnpickler`? Unsure what the proper naming convention is here, but I don't usually see `_` in class names.

File: vision/src/autogluon/vision/predictor/predictor.py [hunk 2]
```diff
@@ -697,7 +698,11 @@ def load(cls, path, verbosity=2):
         if os.path.isdir(path):
             path = os.path.join(path, 'image_predictor.ag')
         with open(path, 'rb') as fid:
-            obj = pickle.load(fid)
+            gpu_count = get_gpu_count_all()
+            if gpu_count > 0:
```
Comment (Innixma) [inline vision/src/autogluon/vision/predictor/predictor.py:702]: General question: Should we even be pickling an object with a torch cuda tensor in the first place? What alternative serialization methods are there?

**Review by Innixma — APPROVED** [review]
LGTM! Added a couple minor comments/questions.

File: vision/src/autogluon/vision/predictor/predictor.py (hunk 2 above)
Comment (yinweisu) [reply vision/src/autogluon/vision/predictor/predictor.py:702]: This is actually an awesome question and I had a similar concern as well. Currently, ImagePredictor is saving a single pickle file. This behavior differs from what tabular and text are doing: saving to a directory. Ideally, we should save the estimator with torch.save() method, which saves to a file, along with our ImagePredictor pickle file into a directory. However, this is an api breaking change

File: vision/src/autogluon/vision/utils/pickle.py (hunk 1 above)
Comment (yinweisu) [reply vision/src/autogluon/vision/utils/pickle.py:6]: The two UU's standing together feels weird, that's why I had the underscore.

**Pull request merged**